src/App.js

```diff
@@ -1,23 +1,97 @@
+
+  const getRandomContact = () => {
+    const remainingContacts = contactsData.filter(
+      (contact) => !contacts.includes(contact)
+    );
+    const randomIndex = Math.floor(Math.random() * remainingContacts.length);
+    return remainingContacts[randomIndex];
+  };
+
+  const addRandomContact = () => {
+    const randomContact = getRandomContact();
+    if (randomContact) {
+      const updatedContacts = [...contacts, randomContact];
+      setContacts(updatedContacts);
+    }
+  };
+
+  const removeContact = (id) => {
+    const updatedContacts = contacts.filter((contact) => contact.id !== id);
+    setContacts(updatedContacts);
+  };
+
+  const sortByName = () => {
+    const sortedContacts = [...contacts].sort((a, b) =>
+      a.name.localeCompare(b.name)
+    );
+    setContacts(sortedContacts);
+  };
+
+  const sortByPopularity = () => {
+    const sortedContacts = [...contacts].sort(
+      (a, b) => b.popularity - a.popularity
+    );
+    setContacts(sortedContacts);
+  };
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>IronContacts</h1>
+      <button onClick={addRandomContact}>Add Random Contact</button>
+      <button onClick={sortByName}>Sort by Name</button>
+      <button onClick={sortByPopularity}>Sort by Popularity</button>
+      <table>
+        <thead>
+          <tr>
+            <th>Picture</th>
+            <th>Name</th>
+            <th>Popularity</th>
+            <th>Won an Oscar</th>
+            <th>Won an Emmy</th>
+            <th>Action</th>
+          </tr>
+        </thead>
+        <tbody>
+          {contacts.map((contact, index) => (
+            <tr key={index}>
+              <td>
+                <img
+                  src={contact.pictureUrl}
+                  alt={contact.name}
+                  style={{ width: "50px", height: "50px" }}
+                />
+              </td>
+              <td>{contact.name}</td>
+              <td>{contact.popularity.toFixed(2)}</td>
+              <td>
+                {contact.wonOscar ? (
+                  <span role="img" aria-label="Oscar Trophy">
+                    🏆
+                  </span>
+                ) : null}
+              </td>
+              <td>
+                {contact.wonEmmy ? (
+                  <span role="img" aria-label="Emmy Trophy">
+                    🏆
+                  </span>
+                ) : null}
+              </td>
+              <td>
+                <button onClick={() => removeContact(contact.id)}>
+                  Delete
+                </button>
+              </td>
+            </tr>
```

Comment:
This hole section code be a new component that only handles the information of the contact, in this case you could do something like 
<ContanctInfo  contact={contact}/>